staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/mutating_work_estimator.go

```diff
@@ -69,24 +78,56 @@ func (e *mutatingWorkEstimator) estimate(r *http.Request) WorkEstimate {
 	// we will work on tuning the algorithm later. As a starting point we
 	// we simply assume that processing 1 event takes 1/Nth of a seat for
 	// M milliseconds and processing different events is infinitely parallelizable.
-	// We simply record the appropriate values here and rely on potential
-	// reshaping of the request if the concurrency limit for a given priority
-	// level will not allow to run request with that many seats.
-	//
+	// We simply estimate the total amount of final work for this request based
+	// on the above and rely on potential reshaping of the request if the
+	// concurrency limit for a given priority level will not allow to run
+	// request with that many seats.
+	var finalSeats uint
+	var additionalLatency time.Duration
+
 	// TODO: As described in the KEP, we should take into account that not all
 	//   events are equal and try to estimate the cost of a single event based on
 	//   some historical data about size of events.
-	var finalSeats uint
-	var additionalLatency time.Duration
+	maxSeats := uint(math.Ceil(float64(watchCount) / watchesPerSeat))
+	finalWork := SeatsTimesDuration(float64(maxSeats), eventAdditionalDuration)
 
 	// TODO: Make this unconditional after we tune the algorithm better.
 	//   Technically, there is an overhead connected to processing an event after
 	//   the request finishes even if there is a small number of watches.
 	//   However, until we tune the estimation we want to stay on the safe side
 	//   an avoid introducing additional latency for almost every single request.
 	if watchCount >= watchesPerSeat {
-		finalSeats = uint(math.Ceil(float64(watchCount) / watchesPerSeat))
-		additionalLatency = eventAdditionalDuration
+		finalSeats = maxSeats
+
+		// While processing individual events is highly parallelizable,
+		// our design/implementation have a couple limitations that would make
+		// it highly inefficient:
```

Comment:
I see two problems with the wording here.
1. "it" is supposed to refer to the most recent noun phrase.  In the current text that is "a couple limitations", which is not what you mean.
2. The limitations in the current APF design/implementation do _not_ make the processing ("sending") of notifications inefficient.  APF does not change how notifications are sent.  The issue here is with how APF manages requests to account for that notification sending.

Reply:
reworder